splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyCheckIT.java

```diff
@@ -706,6 +707,36 @@ public void multipleTablesReferencingSameTable() throws Exception {
     }
 
 
+    @Test
+    public void rowHistoryIsReadyCorrectly() throws Exception {
+            new TableCreator(conn)
+                    .withCreate("create table P(col1 int, col2 varchar(2), col3 int, col4 int, primary key (col2, col4))")
+                    .withInsert("insert into P values(?,?,?,?)")
+                    .withRows(rows(row(1, "a", 1, 1)))
+                    .create();
+            new TableCreator(conn)
+                    .withCreate("create table C (col1 int primary key, col2 varchar(2), col3 int, col4 int, constraint CHILD_FKEY foreign key(col2, col3) references P(col2, col4) )")
+                    .create();
+            conn.commit();
+
+        try(Connection c = newNoAutoCommitConnection()) {
+            c.setAutoCommit(false);
+
+            try (Statement statement = c.createStatement()) {
+                statement.executeUpdate("update P set col1 = 42");
+            }
+            Thread.sleep(30 * 1000);
```

Comment:
I don't like this long pause here. Without the pause I think most of the times the next insert will happen before we have a chance to remove the rollback (specially if we have everything set up, the connection, prepared statement, etc), so I'd rather do that than introduce 30 more seconds of runtime on ITs

Reply:
done.